src/actions/actionFinalize.tsx

```diff
@@ -20,11 +21,16 @@ export const actionFinalize: Action = {
       }
       appState.multiElement.shape = null;
     }
+    if (!appState.elementLocked) {
+      resetCursor();
+    }
     return {
       elements: newElements,
       appState: {
         ...appState,
-        elementType: "selection",
+        elementType: appState.elementLocked
+          ? appState.elementType
+          : "selection",
```

Comment:
We actually keep this behavior even if element is locked. `Esc` in this case should reset to selection no matter what.

Reply:
Except, when in `multiElement` mode. In that case, it should just finalize the shape.

Reply:
fixed in https://github.com/excalidraw/excalidraw/pull/746/commits/12d5d9b704d0b297c88c7366e9a35806cdf147b6